core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -872,6 +917,387 @@ export class ExplorationEditor extends BaseUser {
+    } else {
+      await this.page.waitForSelector(paginatorToggler, {visible: true});
+      await this.page.click(paginatorToggler);
+    }
+
+    const PaginatorOptionsSelector = '.mat-select-panel';
+    const optionSelector = `${PaginatorOptionsSelector} mat-option:nth-child(${index})`;
+    await this.page.waitForSelector(optionSelector, {visible: true});
+    await this.clickOn(optionSelector);
+  }
+
+  /**
+   * Checks if the next page of revisions button is in a certain status.
+   * @param {string} status - The status to check for. Can be 'enabled' or 'disabled'.
+   */
+  async expectNextPageOfRevisionsButtonToBe(status: string): Promise<void> {
+    if (status !== 'enabled' && status !== 'disabled') {
+      throw new Error(
+        'Invalid status argument. Expected "enabled" or "disabled".'
+      );
+    }
+
+    const nextPageButton = await this.page.$('.mat-paginator-navigation-next');
+    if (!nextPageButton) {
+      throw new Error('Next page button not found');
+    }
+    const isDisabled = await nextPageButton.evaluate(
+      button => button.getAttribute('disabled') === 'true'
+    );
+    if (
+      (status === 'disabled' && isDisabled) ||
+      (status === 'enabled' && !isDisabled)
+    ) {
+      showMessage(`Next page button is ${status}`);
+      if (status === 'enabled') {
+        await nextPageButton.click();
+      }
+    } else {
+      throw new Error(
+        `Expected next page button to be ${status}, but it was ${isDisabled ? 'disabled' : 'enabled'}`
+      );
+    }
+  }
+
+  /**
+   * Selects two revisions of an exploration for comparison.
+   * @param {number} revisionIndex1 - The index of the first revision to select.
+   * @param {number} revisionIndex2 - The index of the second revision to select.
+   */
+  async selectTwoRevisionsForComparison(
+    revisionIndex1: number,
+    revisionIndex2: number
+  ): Promise<void> {
+    if (revisionIndex1 === revisionIndex2) {
+      throw new Error('Both revision indexes cannot be the same.');
+    }
+
+    await this.page.waitForSelector(firstRevisionDropdown);
+    await this.clickOn(firstRevisionDropdown);
+    const panel1 = '#mat-select-0-panel';
+    await this.page.waitForSelector(
+      `${panel1} mat-option:nth-last-child(${revisionIndex1})`
+    );
+    await this.clickOn(
+      `${panel1} mat-option:nth-last-child(${revisionIndex1})`
+    );
+
+    await this.page.waitForSelector(secondRevisionDropdown);
+    await this.clickOn(secondRevisionDropdown);
+    const panel2 = '#mat-select-2-panel';
+    await this.page.waitForSelector(
+      `${panel2} mat-option:nth-last-child(${revisionIndex2})`
+    );
+    await this.clickOn(
+      `${panel2} mat-option:nth-last-child(${revisionIndex2})`
+    );
+  }
+
+  /**
+   * Verifies if changes are reflected for each property within a single code block.
+   * @param {string[]} properties - The properties to check for.
+   */
+  async expectChangesInProperties(properties: string[]): Promise<void> {
```

Comment:
I think this method should be `private`.

Reply:
done